applications/app/controllers/InteractiveController.scala

```diff
@@ -5,21 +5,35 @@ import common._
 import contentapi.ContentApiClient
 import conf._
 import conf.switches.Switches
+import model.Cached.RevalidatableResult
 import model._
+import play.api.libs.ws.WSClient
 import play.api.mvc._
 import views.support.RenderOtherStatus
+import conf.Configuration.interactive.cdnPath
+import conf.Configuration.environment.isPreview
 
 import scala.concurrent.Future
 
 case class InteractivePage (interactive: Interactive, related: RelatedContent) extends ContentPage {
   override lazy val item = interactive
 }
 
-class InteractiveController(contentApiClient: ContentApiClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
+class InteractiveController(contentApiClient: ContentApiClient, wsClient: WSClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
 
   def renderInteractiveJson(path: String): Action[AnyContent] = renderInteractive(path)
   def renderInteractive(path: String): Action[AnyContent] = Action.async { implicit request => renderItem(path) }
 
+  def renderInteractiveServiceworker(path: String): Action[AnyContent] = Action.async { implicit request =>
+    val stage = if (isPreview) "preview" else "live"
+    val serviceWorkerPath = s"$cdnPath/service-workers/$stage/$path/interactive-service-worker.js"
+    wsClient.url(serviceWorkerPath).get().map { response =>
+      Cached(60) {
```

Comment:
> I would cache it for far more than 1 minutes.
> Devs can use `router-prod.gu-web.net` if they want to bypass the cache

Why not just honour the cache header received from interactive.guim? That way it's just like any other part of an interactive, and control (and responsibility) rests with the interactive developer.

Otherwise, it needs to be something that all members of interactive teams have access to and instructions on how to use 😀. I can imagine a long cache time being a source of pain and confusion if we need to deploy fixes to interactives post-launch, as often happens.

In any case, from the user's perspective a short cache time on the service worker isn't really that bad, since a) requesting the service worker happens in the background while the existing registered service worker is intercepting fetch events (i.e. it needn't get in the way of other requests), and b) they tend to be small.